docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -107,104 +378,224 @@ public class ProtectedResource {
+While `PKCE` is of primary importance to the public OpenID Connect clients (such as the SPA scripts running in a browser), it can also provide an extra level of protection to Quarkus OIDC `web-app` applications which are confidential OpenID Connect clients capable of securely storing the client secret and using it to exchange the code for the tokens.
+
+You can enable `PKCE` for your OIDC `web-app` endpoint with a `quarkus.oidc.authentication.pkce-required` property and a 32 characters long secret, for example:
+
+[source, properties]
+----
+quarkus.oidc.authentication.pkce-required=true
+quarkus.oidc.authentication.pkce-secret=eUk1p7UB3nFiXZGUXi0uph1Y9p34YhBU
+----
+
+If you already have a 32 characters long client secret then `quarkus.oidc.authentication.pkce-secret` does not have to be set unless you prefer to use a different secret key.
+
+The secret key is required for encrypting a randomly generated `PKCE` `code_verifier` while the user is being redirected with the `code_challenge` query parameter to OpenID Connect Provider to authenticate. The `code_verifier` will be decrypted when the user is redirected back to Quarkus and sent to the token endpoint alongside the `code`, client secret and other parameters to complete the code exchange. The provider will fail the code exchange if a `SHA256` digest of the `code_verifier` does not match the `code_challenge` provided during the authentication request.
 
-OpenID Connect Providers may be configured to require the `redirect_uri` parameter to have the same value (e.g. `http://localhost:8080/service/callback`) for all the redirect URLs.
-In such cases, a `quarkus.oidc.authentication.redirect-path` property has to be set, for example, `quarkus.oidc.authentication.redirect-path=/service/callback`, and Quarkus will set the `redirect_uri` parameter to an absolute URL such as `http://localhost:8080/service/callback` which will be the same regardless of the current request URL.
 
-If `quarkus.oidc.authentication.redirect-path` is set but the original request URL has to be restored after the user has been redirected back to a callback URL such as `http://localhost:8080/service/callback` then a `quarkus.oidc.authentication.restore-path-after-redirect` property has to be set to `true` which will restore the request URL such as `http://localhost:8080/service/1`, and so on.
+=== Handling and controlling the life time of authentication
+
+Another important aspect of authentication is to be sure that data it is based on is fresh while not requiring the user to authenticate at every single request.
+There are also situation when a logout is explicitly requested.
+This section discusses the various key notions necessarity to find the right balance for your Quarkus application.
```

Comment:
Would active voice help here?

```suggestion
Consider the following security lifespan factors when securing your Quarkus application:
```

Reply:
Hi @michelle-purcell I think this change changes a bit the meaning of what Emmanuel is saying, `lifespan` is only one of the technical properties of the token. 
```
This section discusses the various key notions necessary to find the right balance for securing your Quarkus application.
```
should work better ?

Reply:
Where possible, we tried to avoid self-referential statements, especially when action is required.

How about ...? 
```suggestion
Use the following key points to find the right balance for securing your Quarkus applications:
```

Reply:
@michelle-purcell This is perfect, thanks